source/behaviors/offClick/offClick.ts

```diff
@@ -29,16 +30,36 @@ export class OffClickDirective implements OnInit, OnDestroy {
 	}
 
 	ngOnInit() {
-		setTimeout(() => {
-			document.addEventListener('click', this.listener);
-		}, 0);
+		if (this.active) {
+			setTimeout(() => {
+				this.addListener();
+			});
+		}
 	}
 
-	ngOnDestroy() {
+	ngOnChanges(changes: SimpleChanges): void {
+		if (changes['active']) {
+			if (changes['active'].currentValue) {
+				this.addListener();
+			} else {
+				this.removeListener();
+			}
+		}
+	}
+
+	addListener(): void {
+		document.addEventListener('click', this.listener);
```

Comment:
Could we change this to use an abstract class and a value-provider, so that document is injected much like we do window else where?